internal/action/process.go

```diff
@@ -0,0 +1,237 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package action
+
+import (
+	"context"
+	"fmt"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/cenkalti/backoff/v4"
+	"github.com/elastic/e2e-testing/internal/common"
+	"github.com/elastic/e2e-testing/internal/deploy"
+	"github.com/elastic/e2e-testing/internal/utils"
+	log "github.com/sirupsen/logrus"
+)
+
+// actionWaitProcess implements operation for waiting on a process status
+type actionWaitProcess struct {
+	service deploy.ServiceRequest
+	deploy  deploy.Deployment
+	opts    map[string]string
+}
+
+// AttachActionWaitProcess action to woit for a process status on *nix like systems
+func AttachActionWaitProcess(deploy deploy.Deployment, service deploy.ServiceRequest, actionOpts map[string]string) deploy.ServiceOperatorAction {
+	return &actionWaitProcess{
+		service: service,
+		deploy:  deploy,
+		opts:    actionOpts,
+	}
+}
+
+// Run executes the command
+func (a *actionWaitProcess) Run(ctx context.Context) (string, error) {
```

Comment:
Ah got the Command pattern here! :) Forget about this part in my comment above! 👏  

>Besides that, because this is very similar to the Command pattern, I'd rename the Attach method to Run(), Exec() or Execute(), with no parameters. All the input parameters could be part of the state of the disposable ServiceOperation as attributes in the struct on creation.